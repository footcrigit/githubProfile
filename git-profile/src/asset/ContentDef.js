import styled from 'styled-components';

export const NavContent = styled.div `
    margin-top : 1%;
    width : 100%;
    height : 30%;
    border-bottom: 1px solid grey; 
    display: flex;
    flex-direction: row;
    justify-content: space-around;
`;
export const OverViewContent = styled.div `
    width : 100%;
    height : 70%;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
`;
export const OverViewContentCol = styled.div `
    width : 100%;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
`;
export const RepoContent = styled.div `
    width : 50%;
    border : .5px solid grey;
    margin-left : 1%;
    margin-top : 1%;
    display: flex;
    flex-direction: column;
`;
export const RepoContentBar = styled.div `
    width : 50%;
    display: flex;
    flex-direction: column;
`;
export const MainContentDiv = styled.div `
    display: flex;
    flex-direction: column;
`;
export const ButtonNav = styled.button `
    background-color: Transparent; 
    border: none;
    text-color: black;
    padding: 15px 32px;
    font-size: 16px;
    border-radius: 4px
`;
