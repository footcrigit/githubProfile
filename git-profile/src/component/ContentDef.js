import React from 'react';
import {NavContent,OverViewContent,MainContentDiv,
        ButtonNav,RepoContent,OverViewContentCol,RepoContentBar} from '../asset/ContentDef';
import { connect } from 'react-redux';

export class ContentDef extends React.Component {
    constructor(props){
        super(props)
        this.state = {
            name : ''
        }
    }
    componentDidMount(){
        this.setState({
            name : this.props.length
        })
    }
    componentDidUpdate(){
        if(this.props.RepoAPIDetails.length > 0){
            this.setState({
                name : this.props.RepoAPIDetails.length
            })
        }
    }
    render(){
        return(
            <MainContentDiv>
                <NavContent> 
                        <ButtonNav>Overview</ButtonNav>
                        <ButtonNav>Repositories</ButtonNav>
                        <ButtonNav>Project</ButtonNav>
                        <ButtonNav>Stars</ButtonNav>
                        <ButtonNav>Followers</ButtonNav>
                        <ButtonNav>Following</ButtonNav>
                </NavContent>
                <OverViewContentCol>
                <OverViewContent>
                        <RepoContentBar>
                            <section>
                                <h4>Popular Repositories</h4>
                            </section>
                        </RepoContentBar>
                    </OverViewContent>

                    {
                        this.props.RepoAPIDetails.name   ? 
                        this.props.RepoAPIDetails.favreport.map((item, index) =>
                            {
                                return(
                                <OverViewContent key = {index}>
                                    <RepoContent>
                                        <section>
                                            <h5>{item.name}</h5>
                                            <p>{item.description}</p>
                                            <p>{item.language}</p>
                                        </section>
                                    </RepoContent>
                                    <RepoContent>
                                        <section>
                                            <h5>{item.namepair}</h5>
                                            <p>{item.descriptionpair}</p>
                                            <p>{item.languagepair}</p>
                                        </section>
                                    </RepoContent>
                                </OverViewContent>
                                )
                                
                            }) : undefined
                    } 
                </OverViewContentCol>
            </MainContentDiv>
        )
    }
}

const mapStateToProps = (state) => ({
    RepoAPIDetails: state.RepoAPIResponse
})

export default connect(mapStateToProps,undefined)(ContentDef);