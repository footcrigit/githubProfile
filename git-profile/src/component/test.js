import React from 'react';
import {NavContent,OverViewContent,MainContentDiv,
        ButtonNav,RepoContent,OverViewContentCol} from '../asset/ContentDef';
import { connect } from 'react-redux';

class ContentDef extends React.Component {
    constructor(props){
        super(props)
    }
    render(){
        return(
            <MainContentDiv>
                <NavContent> 
                        <ButtonNav>Overview</ButtonNav>
                        <ButtonNav>Repositories</ButtonNav>
                        <ButtonNav>Project</ButtonNav>
                        <ButtonNav>Stars</ButtonNav>
                        <ButtonNav>Followers</ButtonNav>
                        <ButtonNav>Following</ButtonNav>
                </NavContent>
                <OverViewContentCol>
                <OverViewContent>
                        <RepoContent>
                            <section>
                                <h3>Popular Repositories</h3>
                            </section>
                        </RepoContent>
                    </OverViewContent>
                    <OverViewContent>
                        <RepoContent>
                            <section>
                                <h3>WWF</h3>
                                <p>details</p>
                                <p>details</p>
                            </section>
                        </RepoContent>
                        <RepoContent>
                            <section>
                                <h3>WWF</h3>
                                <p>details</p>
                                <p>details</p>
                            </section>
                        </RepoContent>
                    </OverViewContent>
                    <OverViewContent>
                    <RepoContent>
                            <section>
                                <h3>WWF</h3>
                                <p>details</p>
                                <p>details</p>
                            </section>
                        </RepoContent>
                        <RepoContent>
                            <section>
                                <h3>WWF</h3>
                                <p>details</p>
                                <p>details</p>
                            </section>
                        </RepoContent>
                    </OverViewContent>
                    <OverViewContent>
                    <RepoContent>
                            <section>
                                <h3>WWF</h3>
                                <p>details</p>
                                <p>details</p>
                            </section>
                        </RepoContent>
                        <RepoContent>
                            <section>
                                <h3>WWF</h3>
                                <p>details</p>
                                <p>details</p>
                            </section>
                        </RepoContent>
                    </OverViewContent>
                </OverViewContentCol>
            </MainContentDiv>
        )
    }
}

const mapStateToProps = (state) => ({
    RepoAPIDetails: state.RepoAPIResponse
})

export default connect(mapStateToProps,undefined)(ContentDef);