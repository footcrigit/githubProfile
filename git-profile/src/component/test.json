{
    RepoAPIResponse: {
      imgurl: 'https://avatars1.githubusercontent.com/u/7427552?v=4',
      name: 'supreetsingh247',
      favreport: [
        {
          name: 'ActionsAngular',
          description: 'Conditional Discounts',
          language: null
        },
        {
          name: 'bidding',
          description: 'Testing a basic bidding application using react redux',
          language: 'JavaScript'
        },
        {
          name: 'first-contributions',
          description: 'ðŸš€âœ¨ Help beginners contribute to open source projects',
          language: 'JavaScript'
        },
        {
          name: 'FreeCodeCamp',
          description: 'The https://FreeCodeCamp.com open source codebase and curriculum. Learn to code and help nonprofits.',
          language: 'JavaScript'
        },
        {
          name: 'Observe',
          description: 'Test app',
          language: 'JavaScript'
        },
        {
          name: 'pravar',
          description: 'Modified template according to need',
          language: 'CSS'
        },
        {
          name: 'react-basic-pluralsight',
          description: 'A basic react redux application using mock api and thunks. Also used react-router, hot module reloading. ',
          language: 'JavaScript'
        },
        {
          name: 'React-Redux-Boilerplate',
          description: null,
          language: 'JavaScript'
        },
        {
          name: 'superslides',
          description: 'A fullscreen, hardware accelerated slider for jQuery.',
          language: 'JavaScript'
        },
        {
          name: 'vedantu',
          description: null,
          language: 'JavaScript'
        },
        {
          name: 'walnut',
          description: null,
          language: 'HTML'
        },
        {
          name: 'wizard',
          description: 'Angular Wizard',
          language: 'CSS'
        }
      ]
    }
  }