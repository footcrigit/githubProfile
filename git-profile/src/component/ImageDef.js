import React from 'react';
import {Img, ImageContentDiv,Input,Button} from '../asset/ImageDef';
import { connect } from 'react-redux';
import { RepoAPIResponse } from '../action/repoaction';
import {getRepoDetails} from '../service/getRepoDetails';

export class ImageDef extends React.Component {
    constructor(props){
        super(props);
        this.SearchRepo = this.SearchRepo.bind(this);
        this.ApiAction = {
            imgurl:'',
            name:'',
            favreport : []
        };
    }
    async SearchRepo(e){
        e.preventDefault();
        let reponame = e.target.searchRepoID.value;
        let resp = await getRepoDetails(reponame);
        this.props.RepoAPIResponse(resp);
    }
    render(){
        return(
            <ImageContentDiv>
                <form onSubmit = {this.SearchRepo}>
                    <Input type = "text" id = "searchRepoID" placeholder ="Enter Github id" />
                    <Button type = "submit" id ="searchSubmit">Search</Button>
                </form>
                <Img src = {this.props.RepoAPIDetails.imgurl}></Img>
                <h3>{this.props.RepoAPIDetails.name}</h3>
            </ImageContentDiv>
        )
    }
}

const mapDispatchToProps = (dispatch,props) => ({
    RepoAPIResponse : (resp) => dispatch(RepoAPIResponse(resp))
})

const mapStateToProps = (state) => ({
    RepoAPIDetails: state.RepoAPIResponse
})

export default connect(mapStateToProps,mapDispatchToProps)(ImageDef);