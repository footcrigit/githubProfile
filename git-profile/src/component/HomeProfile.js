import React from 'react';
import ImageDef from './ImageDef';
import ContentDef from './ContentDef';
import {getRepoDetails} from '../service/getRepoDetails';
import { connect } from 'react-redux';
import { RepoAPIResponse } from '../action/repoaction';

export class HomeProfile extends React.Component {
    constructor(props){
        super(props)
    }
    async componentDidMount(){
        let response = await getRepoDetails("supreetsingh247");
        this.props.RepoAPIResponse(response);
    }
    render(){
        return(
            <div className = "containerMain">
               <table className = "mainTable">
                   <tbody>
                       <tr >
                            <td className = "imageDiv">
                                <ImageDef></ImageDef>
                            </td>
                            <td className = "ConentDiv">
                                <ContentDef ></ContentDef>
                            </td>
                       </tr>
                   </tbody>
               </table>
            </div>
        )
    }
}
const mapDispatchToProps = (dispatch,props) => ({
    RepoAPIResponse : (resp) => dispatch(RepoAPIResponse(resp))
})


export default connect(undefined,mapDispatchToProps)(HomeProfile);