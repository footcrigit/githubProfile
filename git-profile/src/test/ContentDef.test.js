import React from 'react';
import { shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ContentDef } from '../component/ContentDef';
Enzyme.configure({adapter: new Adapter()})
var props = {
    RepoAPIDetails : {
        "name" : "test",
        "favreport" : [
            {
                "name": "test",
                "description" : "test",
                "language" : "test",
                "namepair": "test1",
                "descriptionpair" : "test2",
                "languagepair" : "test3",
            }
        ]
    }
}
test('should render Homeprofile page', () => {
    const wrapper = shallow (<ContentDef {...props}/>);
    expect(wrapper).toMatchSnapshot();
})