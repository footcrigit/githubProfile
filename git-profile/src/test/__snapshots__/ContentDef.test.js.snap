// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Homeprofile page 1`] = `
<styled.div>
  <styled.div>
    <styled.button>
      Overview
    </styled.button>
    <styled.button>
      Repositories
    </styled.button>
    <styled.button>
      Project
    </styled.button>
    <styled.button>
      Stars
    </styled.button>
    <styled.button>
      Followers
    </styled.button>
    <styled.button>
      Following
    </styled.button>
  </styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        <section>
          <h4>
            Popular Repositories
          </h4>
        </section>
      </styled.div>
    </styled.div>
    <styled.div
      key="0"
    >
      <styled.div>
        <section>
          <h5>
            test
          </h5>
          <p>
            test
          </p>
          <p>
            test
          </p>
        </section>
      </styled.div>
      <styled.div>
        <section>
          <h5>
            test1
          </h5>
          <p>
            test2
          </p>
          <p>
            test3
          </p>
        </section>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
